### PR TITLE
Transpile jsdoc parsing mode

### PR DESCRIPTION
**Pull request by sheetalkamat**
Fixes https://github.com/microsoft/TypeScript/issues/56625

### CODE REVIEW

File: src/services/transpile.ts [hunk 1]
```diff
@@ -121,7 +122,11 @@ export function transpileModule(input: string, transpileOptions: TranspileOption
             languageVersion: getEmitScriptTarget(options),
             impliedNodeFormat: getImpliedNodeFormatForFile(toPath(inputFileName, "", compilerHost.getCanonicalFileName), /*packageJsonInfoCache*/ undefined, compilerHost, options),
             setExternalModuleIndicator: getSetExternalModuleIndicator(options),
-            jsDocParsingMode: JSDocParsingMode.ParseNone,
+            jsDocParsingMode: transpileOptions.jsDocParsingMode !== undefined ?
```
Comment (jakebailey) [inline src/services/transpile.ts:125]: Nit, but it took me three rereads to read this code; can we write it match/case style? Or use `??` somewhere?

**Comment by jakebailey** [comment]
I will be interested in a test for this; my impression was that this flag did not affect emit at all, and #56625 did not mention transformers in the original report.

File: src/services/transpile.ts [hunk 2]
```diff
@@ -121,7 +122,11 @@ export function transpileModule(input: string, transpileOptions: TranspileOption
             languageVersion: getEmitScriptTarget(options),
             impliedNodeFormat: getImpliedNodeFormatForFile(toPath(inputFileName, "", compilerHost.getCanonicalFileName), /*packageJsonInfoCache*/ undefined, compilerHost, options),
             setExternalModuleIndicator: getSetExternalModuleIndicator(options),
-            jsDocParsingMode: JSDocParsingMode.ParseNone,
+            jsDocParsingMode: transpileOptions.jsDocParsingMode !== undefined ?
+                transpileOptions.jsDocParsingMode :
+                transpileOptions.transformers ?
+                JSDocParsingMode.ParseAll :
```
Comment (andrewbranch) [inline src/services/transpile.ts:128]: I think this is a somewhat unintuitive dependent default. I would personally opt for the simplicity of just having the default be `ParseNone`.

Comment (sheetalkamat) [reply src/services/transpile.ts:128]:
```suggestion
```

File: src/services/transpile.ts [hunk 3]
```diff
@@ -121,7 +122,9 @@ export function transpileModule(input: string, transpileOptions: TranspileOption
             languageVersion: getEmitScriptTarget(options),
             impliedNodeFormat: getImpliedNodeFormatForFile(toPath(inputFileName, "", compilerHost.getCanonicalFileName), /*packageJsonInfoCache*/ undefined, compilerHost, options),
             setExternalModuleIndicator: getSetExternalModuleIndicator(options),
-            jsDocParsingMode: JSDocParsingMode.ParseNone,
+            jsDocParsingMode: transpileOptions.jsDocParsingMode !== undefined ?
+                transpileOptions.jsDocParsingMode :
+                JSDocParsingMode.ParseNone,
```
Comment (sheetalkamat) [inline src/services/transpile.ts:127]:
```suggestion
            jsDocParsingMode: transpileOptions.jsDocParsingMode ?? JSDocParsingMode.ParseNone,
```

**Review by andrewbranch — APPROVED** [review]

**Comment by andrewbranch** [comment]
@typescript-bot cherry-pick this to release-5.3

File: src/services/transpile.ts (hunk 2 above)
Comment (jakebailey) [reply src/services/transpile.ts]: To be honest, I would have made this default to `ParseAll`; when I wrote this I had forgotten about transformers and assumed that this was pure string to string without them.

**Pull request merged**